nav2_util/include/nav2_util/service_client.hpp

```diff
@@ -137,6 +173,7 @@ class ServiceClient
   rclcpp::Node::SharedPtr node_;
   rclcpp::CallbackGroup::SharedPtr callback_group_;
   rclcpp::executors::SingleThreadedExecutor callback_group_executor_;
```

Comment:
The executor should be made into a unique pointer. The executor's constructor will call the abstract executor's constructor https://github.com/ros2/rclcpp/blob/master/rclcpp/src/rclcpp/executor.cpp which does a bunch of set up and connects to the network. We'd like to avoid that if possible unless its actually going to be used. So if its a unique pointer, then it will be `nullptr` and not create an object until you create one `make_unique` when it may be used.